core/domain/exp_jobs_one_off.py

```diff
@@ -831,3 +832,100 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class ExplorationMigrationValidationJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job to validate exploration migration can be carried out
+    successfully.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted or item.states_schema_version != (
+                feconf.CURRENT_STATES_SCHEMA_VERSION):
+            return
+
+        try:
+            exploration = exp_services.get_exploration_from_model(item)
+            exp_rights = rights_manager.get_exploration_rights(item.id)
+        except Exception as e:
+            yield('Error %s when loading exploration' % str(e), item.id)
+            return
+
+        try:
+            if exp_rights.status == rights_manager.ACTIVITY_STATUS_PRIVATE:
+                exploration.validate()
+            else:
+                exploration.validate(strict=True)
+        except Exception as e:
+            yield('Error %s when validating current exploration' % str(e),
+                  item.id)
+            return
+
+
+        states = exploration.states
+        new_states = {}
+        for state_name, state in states.iteritems():
+            state_dict = state.to_dict()
+            state_content_id_list = []
+
+            # Add state card's content id into the state_content_id_list.
+            state_content_id_list.append(state_dict['content']['content_id'])
+
+            # Add answer_groups content id into the state_content_id_list.
+            for answer_group in state_dict['interaction']['answer_groups']:
+                answer_feedback = answer_group['outcome']['feedback']
+                state_content_id_list.append(answer_feedback['content_id'])
+
+            # If present, add default_outcome content id into
+            # state_content_id_list.
+            default_outcome = state_dict['interaction']['default_outcome']
+            if default_outcome is not None:
+                state_content_id_list.append(
+                    default_outcome['feedback']['content_id'])
+
+            # Add hints content id into state_content_id_list.
+            for hint in state_dict['interaction']['hints']:
+                state_content_id_list.append(hint['hint_content']['content_id'])
+
+            # If present, add solution content id into state_content_id_list.
+            solution = state_dict['interaction']['solution']
+            if solution:
+                state_content_id_list.append(
+                    solution['explanation']['content_id'])
+
+            # Filter content_ids_to_audio_translations with unwanted
```

Comment:
Do you want to yield any info/data about the items removed here (I know you said it's OK to remove them, but just to be sure / check your assumptions)?

Reply:
I don't think this data will be of any use? (This can only tell us that we are removing something which was left here of no use and if we found some data out of it then we can remove those data from GCS!)

Okay, I'll add a yield over here!

Reply:
But you're assuming it'll be empty, right? If so, might want to add something that confirms that. (If not, what is the basis for saying "no use"?)

Reply:
I'm assuming it to be empty but it can have something which is of no use because of our current exploration validation check.

This can have a content_id which isn't present within the state. and In the worst case having audio for the same content_id is of no use as it doesn't have any reference to the text/HTML.

Reply:
Then maybe validate that -- i.e. for any fields that are deleted, log the value and verify that it's either empty or has a content ID not in the state etc.

It might be interesting to examine those cases and find out why they happened. And if there are none, we have an additional layer of confidence.

Reply:
I'll add a log for the content_ids we are deleting from the content_ids_to_audio_translations. 
> verify that it's either empty or has a content ID, not in the state

 We are creating this `extra_content_ids_in_citat` by set(content_ids_to_audio_translations.keys()) - set(all_content_ids_present_in_the_state) over [here](https://github.com/oppia/oppia/pull/6249/files#diff-e4833416cf9d665369010673d2fba853R905), I'm not sure whether I have to reverify that the content ids present in the `extra_content_ids_in_citat` isn't present in the entire state.

Reply:
Done!